receiver/mongodbatlasreceiver/internal/metadata/generated_metrics.go

```diff
@@ -0,0 +1,580 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by mdatagen. DO NOT EDIT.
+
+package metadata
+
+import (
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+// Type is the component type name.
+const Type config.Type = "mongoatlasreceiver"
+
+// MetricIntf is an interface to generically interact with generated metric.
+type MetricIntf interface {
+	Name() string
+	New() pdata.Metric
+	Init(metric pdata.Metric)
+}
+
+// Intentionally not exposing this so that it is opaque and can change freely.
+type metricImpl struct {
```

Comment:
Not a comment on the substance of metrics produced, but I wonder if it would be easier to manage a slice of structs with name, description, unit, and datatype fields that's used by a metric producer.  If necessary the struct could have an initializer function field that's used for unique items.  If `MetricIntf` has a larger intended usage then maybe the reduced redundancy isn't worth it though.

Reply:
Yeah, I keep getting the urge to do that, but since many of these metricImpl instances match to a single pdata.Metric instance, I'd end up duplicating the data. I might be able to build a different struct that did a better job of keeping it all together though.